### ISSUE TITLE
New Rule: filename is valid accessible name

### ISSUE THREAD

**Issue by jeeyyy** [dequelabs/axe-core#1504]
# filename-is-valid-accessible-name

This rule checks that image elements that use their source filename as their accessible name do so without loss of information to the user.

## Tags
- wcag2
- wcag111
- experimental

## Matches

- any HTML input element with a type of image
- any HTML element with the semantic role of img, 
- is not `aria-hidden`.
- has `accessible name`.

## Checks

**all:**
- filename-is-valid-accessible-name
	- `incomplete` 
		- `Element does not have an accessible name that serves an equivalent purpose to the non-text content`.
	- `pass`: 
		- `Element has an accessible name that serves an equivalent purpose to the non-text content`.

### References:
- [ACT Rule - Filename is valid accessible name](https://act-rules.github.io/rules/9eb3f6)

**Comment by dylanb** [comment]
I suggest that we close this because axe-core does not and cannot infer accurate meaning at this point in time i.e. this rule proposal is out of scope for axe-core

**Comment by jkodu** [comment]
@dylanb, we intend to only return `incomplete` on this rule and not a `failure`. Perhaps can mark this rule as `experimental` too.

**Comment by dylanb** [comment]
ok, I am worried about noise

**Comment by jkodu** [comment]
# Test cases (WIP)

```html
<!-- Pass 1, accessible name, uses filename -->
<img src="https://www.w3.org/WAI/demos/bad/img/w3c" alt="w3c" />

<!-- Pass 2, accessible name, uses filename (along with extension) -->
<img src="https://www.w3.org/WAI/demos/bad/img/w3c.png" alt="Download w3c.png"
/>

<!-- Incomplete 1, accessible name matches file name, but presence of extension in accessible name results in accesible name not describing the image accurately -->
<img src="https://www.w3.org/WAI/demos/bad/img/w3c.png" alt="w3c.png" />

<!-- Incomplete 2, same as above, but different element type -->
<input type="image" src="https://www.w3.org/WAI/demos/bad/before/img/top_weather.gif" alt="top_weather.gif" />

<!-- Inapplicable 1 -->
<img role="presentation" />

<!-- Inapplicable 2 -->
<img style="display:none;" />
```

**Comment by jkodu** [comment]
@jeankaplansky the rule has been merged #1649 and is ready for docs.

**Comment by WilcoFiers** [comment]
Closing this. ACT has deprecated this rule.
